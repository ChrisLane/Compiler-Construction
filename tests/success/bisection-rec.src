// This is the recursive bisection algorithm. This should parse.
bisect (l, r) {
    int mid = (r - l) / 2;
    
    if (mid >= E) {
        if (f(r) * f(mid) <= 0) {
            l = mid;
        } else {
            r = mid;
        };
        bisect(l, r);
    } else {};

    return mid;
}
