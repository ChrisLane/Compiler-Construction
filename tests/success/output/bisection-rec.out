bisect ( l, r ) { 
    New ( "mid" = Operator ( Divide, Operator ( Minus, "r" , "l" ) , Const 2 ) ) In { 
        If ( Operator ( Geq, "mid" , "E" ) ) { 
            If ( Operator ( Leq, Operator ( Times, Application ( "f" ( "r" ) , Application ( "f" ( "mid" ) ) , Const 0 ) ) { 
                Asg ( "l" := "mid" ) 
            } Else { 
                Asg ( "r" := "mid" ) 
            } ; 
            Application ( "bisect" ( "l" , "r" ) 
        } Else { 
            empty 
        } ; 
        Deref ( "mid" ) 
    } 
}
